### PR TITLE
Support UI5 interfaces extending classes or other interfaces

### PR DESCRIPTION
**Pull request by codeworrior**

### CODE REVIEW

**Comment by bd82** [comment]
```
[warn] packages/dts-generator/lib/phases/json-to-ast.js
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
```

**Comment by bd82** [comment]
HI @codeworrior 

Would this change affect the generation of one of the libraries here:
- https://github.com/SAP/ui5-typescript/tree/master/test-packages/openui5-snapshot-test/output-dts ?

If so you also need to run the `re-generate` script in the snapshot tests, otherwise the build would fail.
- https://github.com/SAP/ui5-typescript/blob/master/test-packages/openui5-snapshot-test/package.json#L21

**Comment by bd82** [comment]
Commit message needs to follow conventional commit angular notation.
- https://github.com/SAP/ui5-typescript/blob/master/CONTRIBUTING.md#commit-messages-format

This is later used to auto-generate the CHANGELOGs
- https://github.com/SAP/ui5-typescript/blob/master/packages/dts-generator/CHANGELOG.md

I am not sure if this is a `fix` or a `feat` (new feature).

File: packages/dts-generator/lib/phases/json-to-ast.js [hunk 1]
```diff
@@ -280,6 +280,13 @@ function buildClass(ui5Class) {
   return astNode;
 }
 
+const ui5ExtendsToArray = (ui5Extends) => {
+	if ( Array.isArray(ui5Extends) ) {
```
Comment (bd82) [inline packages/dts-generator/lib/phases/json-to-ast.js:284]: This project uses `lodash` so lets use _.isArray and _.clone.
The second one is also clearer than `slice` as the verb is closer to the intent.

**Comment by codeworrior** [comment]
> ```
> [warn] packages/dts-generator/lib/phases/json-to-ast.js
> [warn] Code style issues found in the above file(s). Forgot to run Prettier?
> ```
> 
> 
> Prettier is normally executed on a pre-commit hook.
> 
> So maybe you made those changes without first running `yarn` so the dev-deps were not installed?

I made the changes via Github, yes.

File: packages/dts-generator/lib/phases/json-to-ast.js (hunk 1 above)
Comment (codeworrior) [reply packages/dts-generator/lib/phases/json-to-ast.js]: Done

**Comment by bd82** [comment]
@codeworrior does the interface which extends appears (usage) in any of the openui5 libs?
I could update the snapshot test in that case so we would have a better idea of how this change effects code generation.

For example in this PR you can see how a d.ts file changed due to the dts-generator changes.
- https://github.com/SAP/ui5-typescript/pull/208/files

**Comment by codeworrior** [comment]
Locally, I did such a test with `IFormContent`, but that's not part of the committed sources yet. 
I'm collection information from colleagues and then will update at least some interfaces. But for now, none of the interfaces has an `extends`.

I thought, the gap is quite obvious and we can fix it already. But when you prefer to have 'living' samples first, we can for sure wait.

**Comment by bd82** [comment]
> I thought, the gap is quite obvious and we can fix it already. But when you prefer to have 'living' samples first, we can for sure wait.

We don't have to wait, I was just checking if we can add another library or two to the snapshot tests so
we could have a "complete" E2E test in this PR's scope.

**Pull request merged**